ovs/dal/dataobjectlist.py

```diff
@@ -273,3 +281,13 @@ def pop(self, index):
             raise ValueError('Index must be an integer')
         self._guids.pop(index)
         self._objects = dict(item for item in self._objects.iteritems() if item[0] in self._guids)
+
+    def shuffle(self):
+        """
+        Randomly shuffle the items in the data-object list
+        :return: Shuffled data-object list
+        """
+        self.load()
+        objects = [self._objects[guid] for guid in self._guids]
+        random.shuffle(objects)
+        self._guids = [obj.guid for obj in objects]
```

Comment:
Can't you just shuffle the guids? The iterators use the order of the guids anyway and you don't have to load all objects.